projects/plugins/jetpack/modules/sso.php

```diff
@@ -1126,6 +1131,30 @@ public function is_user_connected( $user_id ) {
 	public function get_user_data( $user_id ) {
 		return get_user_meta( $user_id, 'wpcom_user_data', true );
 	}
+
+	/**
+	 * Check if the site has a custom login page URL, and return it.
+	 * If default login page URL is used (`wp-login.php`), `null` will be returned.
+	 *
+	 * @return string|null
+	 */
+	private function get_custom_login_url() {
+		$login_url = wp_login_url();
+
+		if ( 'wp-login.php' === substr( $login_url, 0, -12 ) ) {
+			// No custom URL found.
+			return null;
+		}
+
+		$site_url = trailingslashit( site_url() );
+
+		if ( 0 !== strpos( $login_url, $site_url ) ) {
+			// Something went wrong, we can't properly extract the custom URL.
+			return null;
+		}
+
+		return str_ireplace( $site_url, '', $login_url );
```

Comment:
Could you explain a bit why we need to do this? I'm not sure I understand.

Reply:
Hi @jeherve, thanks for the review!
Good point about the unit test, I added it.

> Could you explain a bit why we need to do this? I'm not sure I understand.

The `wp_login_url()` function returns the full URL, including the domain name (e.g. `https://example.org/custom-login-page/`). We only need the `custom-login-page/` part of this, because it wouldn't be quite safe to pass the full URL to redirect the user to.

So first we check if we can actually extract that part by by removing the `site_url` from the beginning of the string (the `strpos()` condition).
If we can do that, we use `str_ireplace` to actually remove it.
In the end, function returns `custom-login-page/` only, which gets sent to WP.com